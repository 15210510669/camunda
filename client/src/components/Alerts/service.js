import {get, post, del, put} from 'request';

export async function loadAlerts() {
  const response = await get('/api/alert');

  return await response.json();
}

export async function loadReports() {
  const response = await get('/api/report');

  return await response.json();
}

export async function saveNewAlert(alert) {
  const response = await post('/api/alert', alert);

  const json = await response.json();

  return json.id;
}

export async function updateAlert(id, alert) {
  return await put('/api/alert/' + id, alert);
}

export async function deleteAlert(id) {
  return await del('api/alert/' + id);
}

