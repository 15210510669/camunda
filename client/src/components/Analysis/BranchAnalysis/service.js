/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import {post} from 'request';

export async function loadFrequencyData(
  processDefinitionKey,
  processDefinitionVersions,
  tenantIds,
  filter
) {
  const response = await post(
    'api/report/evaluate',
    createFlowNodeFrequencyReport(
      processDefinitionKey,
      processDefinitionVersions,
      tenantIds,
      filter
    )
  );

  return await response.json();
}

function createFlowNodeFrequencyReport(
  processDefinitionKey,
  processDefinitionVersions,
  tenantIds,
  filter
) {
  return {
    combined: false,
    reportType: 'process',
    data: {
      processDefinitionKey,
      processDefinitionVersions,
      tenantIds,
      filter,
      view: {
        entity: 'flowNode',
        property: 'frequency',
      },
      groupBy: {
        type: 'flowNodes',
        unit: null,
      },
      visualization: 'heat',
    },
  };
}

export async function loadCorrelationData(
  processDefinitionKey,
  processDefinitionVersions,
  tenantIds,
  filter,
  gateway,
  end
) {
  const response = await post('api/analysis/correlation', {
    processDefinitionKey,
    processDefinitionVersions,
    tenantIds,
    filter,
    gateway,
    end,
  });

  return await response.json();
}
