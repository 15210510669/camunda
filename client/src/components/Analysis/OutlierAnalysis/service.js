/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import {get} from 'request';

export async function loadNodesOutliers(config) {
  const response = await get('api/analysis/flowNodeOutliers', config);
  return await response.json();
}

export {loadDurationData, loadCommonOutliersVariables, getInstancesDownloadUrl} from './service.ts';
