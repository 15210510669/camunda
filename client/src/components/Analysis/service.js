import {get, post} from 'request';

export async function loadProcessDefinitionXml(processDefinitionKey, processDefinitionVersion) {
  const response = await get('/api/process-definition/xml', {processDefinitionKey, processDefinitionVersion});

  return await response.text();
}

export async function loadFrequencyData(processDefinitionKey, processDefinitionVersion, filter) {
  const response = await post('/api/process-definition/heatmap/frequency', {
    processDefinitionKey,
    processDefinitionVersion,
    filter
  });

  return await response.json();
}

export async function loadCorrelationData(processDefinitionKey, processDefinitionVersion, filter, gateway, end) {
  const response = await post('/api/process-definition/correlation', {
    processDefinitionKey,
    processDefinitionVersion,
    filter,
    gateway,
    end
  });

  return await response.json();
}
