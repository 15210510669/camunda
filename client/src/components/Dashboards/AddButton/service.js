/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import {get} from 'request';

export async function loadReports() {
  const response = await get('api/entities');
  const entities = await response.json();

  return filterReports(entities);
}

export async function loadReportsInCollection(id) {
  const response = await get('api/collection/' + id);
  const collection = await response.json();

  return filterReports(collection.data.entities);
}

function filterReports(data) {
  return data.filter(({entityType}) => entityType === 'report');
}
