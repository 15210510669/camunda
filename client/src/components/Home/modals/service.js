/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import {get} from 'request';
import {t} from 'translation';

export async function searchIdentities(terms) {
  const response = await get(`api/identity/search`, {terms});

  return await response.json();
}

export async function getDefinitionTenants(defintionKey, defintionType) {
  const response = await get(`api/definition/${defintionType}/${defintionKey}`);

  return await response.json();
}

export async function getDefinitionsWithTenants() {
  const response = await get('api/definition');

  return await response.json();
}

export async function getTenantsWithDefinitions() {
  const response = await get('api/definition/_groupByTenant');

  return await response.json();
}

export function formatDefintionName({name, type}) {
  const typeLabel = type === 'process' ? t('home.sources.process') : t('home.sources.decision');

  return name + ' (' + typeLabel + ')';
}
