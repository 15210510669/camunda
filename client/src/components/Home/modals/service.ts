/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import {get} from 'request';
import {Definition, Tenant} from 'types';

export type TenantWithDefinitions = Tenant & {definitions: Definition};

export type DefinitionWithTenants = Omit<Definition, 'key'> & {tenants: Tenant[]; key: string};

export async function getDefinitionsWithTenants(): Promise<DefinitionWithTenants[]> {
  const response = await get('api/definition');

  return await response.json();
}

export async function getTenantsWithDefinitions(): Promise<TenantWithDefinitions[]> {
  const response = await get('api/definition/_groupByTenant');

  return await response.json();
}
