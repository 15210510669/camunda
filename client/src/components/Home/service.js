/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import {get, post, put, del} from 'request';

export async function loadEntities() {
  const response = await get('api/entities');
  return await response.json();
}

export async function addUser(collection, id, type, role) {
  return await post(`api/collection/${collection}/role`, {identity: {id, type}, role});
}

export async function editUser(collection, id, role) {
  return await put(`api/collection/${collection}/role/${id}`, {role});
}

export async function removeUser(collection, id) {
  return await del(`api/collection/${collection}/role/${id}`);
}

export async function loadAlerts(collection) {
  const response = await get(`api/collection/${collection}/alerts`);
  return await response.json();
}

export async function addAlert(alert) {
  return await post(`api/alert`, alert);
}

export async function editAlert(id, alert) {
  return await put(`api/alert/${id}`, alert);
}

export async function removeAlert(id) {
  return await del(`api/alert/${id}`);
}

export async function copyEntity(type, id, name, collectionId) {
  const query = {name};

  if (collectionId || collectionId === null) {
    query.collectionId = collectionId;
  }

  const response = await post(`api/${type}/${id}/copy`, undefined, {query});
  const json = await response.json();

  return json.id;
}

export async function searchIdentities(terms) {
  const response = await get(`api/identity/search`, {terms});

  return await response.json();
}
