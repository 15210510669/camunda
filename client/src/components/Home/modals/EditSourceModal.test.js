/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import React from 'react';
import {shallow} from 'enzyme';
import {getDefinitionTenants} from './service';

import EditSourceModalWithErrorHandling from './EditSourceModal';

const EditSourceModal = EditSourceModalWithErrorHandling.WrappedComponent;

jest.mock('./service', () => ({getDefinitionTenants: jest.fn()}));

const props = {
  mightFail: jest.fn().mockImplementation((data, cb) => cb(data)),
  onClose: jest.fn(),
  onConfirm: jest.fn(),
  source: {
    definitionKey: 'defKey',
    definitionName: 'defName',
    tenants: [
      {id: null, name: 'Not defined'},
      {id: '__unauthorizedTenantId__', name: 'unauthorizedTenant'},
    ],
  },
};

getDefinitionTenants.mockReturnValue({
  ...props.source,
  tenants: [
    {id: null, name: 'Not defined'},
    {id: 'test', name: 'testName'},
  ],
});

it('should match snapshot', () => {
  const node = shallow(<EditSourceModal {...props} />);

  expect(node).toMatchSnapshot();
});

it('should get defintion tenants on mount', () => {
  shallow(<EditSourceModal {...props} />);

  expect(getDefinitionTenants).toHaveBeenCalled();
});

it('should update selected tenants on itemList change', () => {
  const node = shallow(<EditSourceModal {...props} />);

  node
    .find('Checklist')
    .props()
    .onChange([{id: null, name: 'Not defined'}]);

  node.find('.confirm').simulate('click');

  expect(props.onConfirm).toHaveBeenCalledWith([null]);
});

it('should not deselect unauthorized tenants', () => {
  const node = shallow(<EditSourceModal {...props} />);

  node.find('Checklist').props().onChange([]);

  expect(node.find('Checklist').props().selectedItems).toEqual([
    {id: '__unauthorizedTenantId__', name: 'unauthorizedTenant'},
  ]);
});
