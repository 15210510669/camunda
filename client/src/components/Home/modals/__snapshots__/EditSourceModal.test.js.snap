// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<Modal
  className="EditSourceModal"
  onClose={[MockFunction]}
  onConfirm={[Function]}
  open={true}
>
  <Component>
    Edit defName
  </Component>
  <Component>
    <Form>
      Tenants
      <FormGroup>
        <ItemsList
          allItems={
            Array [
              Object {
                "id": null,
                "name": "Not defined",
              },
              Object {
                "id": "test",
                "name": "testName",
              },
              Object {
                "id": "__unauthorizedTenantId__",
                "name": "unauthorizedTenant",
              },
            ]
          }
          onChange={[Function]}
          selectedItems={
            Array [
              Object {
                "id": null,
                "name": "Not defined",
              },
              Object {
                "id": "__unauthorizedTenantId__",
                "name": "unauthorizedTenant",
              },
            ]
          }
        />
      </FormGroup>
    </Form>
  </Component>
  <Component>
    <ForwardRef(Button)
      className="cancel"
      onClick={[MockFunction]}
    >
      Cancel
    </ForwardRef(Button)>
    <ForwardRef(Button)
      className="confirm"
      disabled={false}
      onClick={[Function]}
      primary={true}
    >
      Apply
    </ForwardRef(Button)>
  </Component>
</Modal>
`;
